catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoListener.java

```diff
@@ -164,11 +155,28 @@ public int getAnalogPin5() {
 		return analogPin5;
 	}
 
-	public int getPortValue(int pin) {
-		return portValue[pin];
+	public int getPinValue(int pin) {
+		if ((pin >= 0) && (pin < ArduinoImpl.NUMBER_OF_DIGITAL_PINS)) {
+			int port = (pin + 7) / 8;
+			int index = pin % 8;
+			return ArduinoImpl.getBit(this.portValue[port], index);
+		}
+		return 0;
+	}
+
+	public void setPinValue(int pin, int value) {
+		if ((pin >= 0) && (pin < ArduinoImpl.NUMBER_OF_DIGITAL_PINS)) {
+			int port = (pin + 7) / 8;
+			int index = pin % 8;
+			this.portValue[port] = ArduinoImpl.setBit(this.portValue[port], index, value);
+		}
```

Comment:
ok. Which of the following styles do you prefer:
`public int getPinValue(int pin) {
		if (ArduinoImpl.isValidPin(pin)) {
			return Utils.getBit(portValue[ArduinoImpl.getPortFromPin(pin)], ArduinoImpl.getIndexOfPinOnPort(pin));
		}
		return 0;
	}

	public void setPinValue(int pin, int value) {
		if (ArduinoImpl.isValidPin(pin)) {
			int port = ArduinoImpl.getPortFromPin(pin);
			int index = ArduinoImpl.getIndexOfPinOnPort(pin);
			this.portValue[port] = Utils.setBit(portValue[port], index, value);
		}
	}`